test/promise.cpp

```diff
@@ -10,6 +10,123 @@
 
 namespace {
 
+template<size_t Alignment, size_t Size = 4096>
```

Comment:
I think it's betttter to move allocator and arena into separate header in the test directory. It's always better to see test code as close to the top of the file as possible so I prefere to move complex helper classes into separate files. Additional benifit: the same allocator will be reused for packaged_task allocator support tests later.